src/publishers/tools.py

```diff
@@ -20,6 +20,97 @@
 logger = get_logger(__name__)
 
 
+def load_collections_file(
+    file_path: str | Path,
+) -> list[dict[str, str]] | None:
+    """Load collections from YAML or JSON file.
+
+    Args:
+        file_path: Path to collections file (YAML or JSON)
+
+    Returns:
+        List of collection dicts with 'name' and optional 'version',
+        or None if file doesn't exist
+
+    Raises:
+        TypeError: If file format is invalid (wrong type)
+        ValueError: If file format is invalid (parse error)
+        OSError: If file cannot be read
+    """
+    file_path_obj = Path(file_path)
+    if not file_path_obj.exists():
+        logger.warning(f"Collections file not found: {file_path}")
+        return None
+
+    try:
+        with file_path_obj.open() as f:
+            if file_path_obj.suffix in [".yaml", ".yml"]:
+                data = yaml.safe_load(f)
+            else:
+                data = json.load(f)
+    except (yaml.YAMLError, json.JSONDecodeError) as e:
+        error_msg = f"Failed to parse collections file {file_path}: {e}"
+        logger.error(error_msg)
+        raise ValueError(error_msg) from e
+    except Exception as e:
+        error_msg = f"Failed to load collections file {file_path}: {e}"
+        logger.error(error_msg)
+        raise OSError(error_msg) from e
```

Comment:
I would use RuntimeError, OsError does not seems great here, it's already open, no?